src/request.ts

```diff
@@ -115,28 +121,34 @@ export class HonoRequest<P extends string = '/', I extends Input['out'] = {}> {
   }
 
   async parseBody<T extends BodyData = BodyData>(): Promise<T> {
-    return await parseBody(this.raw)
+    return await parseBody(this)
+  }
+
+  private cachedBody = (key: keyof BodyCache) => {
+    const cachedBody = this.bodyCache[key]
```

Comment:
I like this a little better for perf since it reduces calls to getters. But maybe less clear?
```ts
    const { bodyCache, raw } = this
    const cachedBody = bodyCache[key]
    if (cachedBody) return cachedBody
    return (bodyCache[key] = raw[key]())
```

Reply:
Ah, you are right. Going with it. Thanks!

Reply:
Wait. I think these are just accessing to props, so they shouldn't affect performance, right?

Reply:
@ThatOneBro 

Thanks for explanation the details! yeah, let's go with your code!